### PR TITLE
feat: test cases for the Tab component

### PR DESCRIPTION
**Pull request by mahesh0618**

### CODE REVIEW

**Comment by zoltan-dulac** [comment]
This is a good start.  I would like the following added:

1) Do the selector fix outlined above.
2) I would also like to add in the unit test that the active tab always has `aria-selected="true"`.
3) Ensure the DOM has the right structure.  Take a look at the code walk-through on the [Enable Tabs page](https://www.useragentman.com/enable/tabs.php).  Steps #5-#9 in that walk-through will tell you what you should test for. (I should have outlined that to you before starting work that you should go through the code walk-throughs).





Let me know when this is done and I'll approve (I do want to run on my local to double check when I am done, but it looks like it works as it should.

**Comment by mahesh0618** [comment]
@zoltan-dulac Feedback has been incorporated, please validate.

File: js/test/tabs.test.js [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+                // Define a list of selectors for focusable elements
+                const focusableSelectors = [
+                    'a[href]',
+                    'button',
+                    'textarea',
+                    'input',
+                    'select',
+                    '[tabindex]:not([tabindex="-1"])',
+                    '[contenteditable]',
+                ];
```
[the remark this answers is not in the document]
Comment (mahesh0618) [reply js/test/tabs.test.js]: the feedback has been incorporated

File: js/test/tabs.test.js [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+            },
+            tabPanel,
+            accessibility.tabbableSelector,
+        );
+
+        if (tabPanelDomInfo.focusableElements.length > 0) {
+            expect(tabPanelDomInfo.isActiveElementInCurrentTabPanel).toBe(true);
+        }
+    });
+});
```
Comment (zoltan-dulac) [inline js/test/tabs.test.js:206]: One thing I did forget: we should have a unit test where we test JS off on the page.  If you test the current page with JS off, it should behave like a bunch of anchors to content and work (albeit differently).  


I think you can do it using this:


```

const page = await browser.newPage()
await page.setJavaScriptEnabled(false)

```

Take a look at [this stack overflow page ](https://stackoverflow.com/questions/51897899/how-to-stop-all-js-scripts-in-puppeteer) for more info.


I have never done this before, but I think this should work, and it should be tested.

File: js/test/tabs.test.js [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+                ) || Array.from(document.querySelectorAll(`${tabSelector}`))[0];
+
+            focusTab.focus();
+
+            return {
+                focusDefaultTab: document.activeElement === focusTab,
+            };
+        }, tabSelector);
+
+        expect(domInfo.focusDefaultTab).toBe(true);
```
Comment (zoltan-dulac) [inline js/test/tabs.test.js:30]: There should be a test that:

1) Focus one interactive element before the tab component.
2) Press the TAB key.
3) See which Tab is accessed.   Is it the selected one?
4) Tab out of it.
5) Tab backwards into it.  Is it is selected tab again?

You should also test whether the instructions appear with keyboard only.

**Review by zoltan-dulac — CHANGES_REQUESTED** [review]

**Comment by zoltan-dulac** [comment]
Hi Mahesh! Good job so far.  Just want a few more changes.  After those are done, I think we can merge.

File: js/test/tabs.test.js [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+
+                    // Convert the NodeList to an Array (if needed) and return it
+                    return Array.from(focusableElements);
+                };
+
+                let focusableElements = [];
+
+                if (tabPanelVisibleEl) {
+                    focusableElements = getFocusableElements(tabPanelVisibleEl);
+                }
```
[the remark this answers is not in the document]
Comment (mahesh0618) [reply js/test/tabs.test.js:194]: updated to use ```contains```

File: js/test/tabs.test.js (hunk 2 above)
Comment (mahesh0618) [reply js/test/tabs.test.js:350]: added a test case to check JS disabled scenarios

File: js/test/tabs.test.js (hunk 3 above)
Comment (mahesh0618) [reply js/test/tabs.test.js:34]: added a test case to check the outlined scenarios

**Review by zoltan-dulac — APPROVED** [review]

**Pull request merged**